jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/MaximumFlowAlgorithm.java

```diff
@@ -0,0 +1,86 @@
+/* ==========================================
+ * JGraphT : a free Java graph-theory library
+ * ==========================================
+ *
+ * Project Info:  http://jgrapht.sourceforge.net/
+ * Project Creator:  Barak Naveh (http://sourceforge.net/users/barak_naveh)
+ *
+ * (C) Copyright 2003-2008, by Barak Naveh and Contributors.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+/* -----------------
+ * MaximumFlowAlgorithm.java
+ * -----------------
+ * (C) Copyright 2015-2015, by Alexey Kudinkin and Contributors.
+ *
+ * Original Author:  Alexey Kudinkin
+ * Contributor(s):
+ *
+ * $Id$
+ *
+ * Changes
+ * -------
+ */
+
+package org.jgrapht.alg.interfaces;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Allows to derive <a
+ * href="https://en.wikipedia.org/wiki/Maximum_flow_problem">maximum-flow</a> from
+ * the supplied <a href="https://en.wikipedia.org/wiki/Flow_network">flow network</a>
+ *
+ * @param <V> vertex concept type
+ * @param <E> edge concept type
+ */
+public interface MaximumFlowAlgorithm<V, E> {
+
+    class MaximumFlow<V, E> {
```

Comment:
I see...my apologies for the confusion; somehow I overlooked that part of the original exchange.  Let's leave it as is for this review pass (I might follow up myself in a later commit).